# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/aubio/hotfix-python.patch
# Copyright (C) 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- aubio-0.4.9/waflib/Context.py.vanilla	2024-05-10 23:22:39.441526444 +0200
+++ aubio-0.4.9/waflib/Context.py	2024-05-10 23:33:39.706455591 +0200
@@ -2,7 +2,7 @@
 # encoding: utf-8
 # WARNING! Do not edit! https://waf.io/book/index.html#_obtaining_the_waf_file
 
-import os,re,imp,sys
+import os,re,types,sys
 from waflib import Utils,Errors,Logs
 import waflib.Node
 HEXVERSION=0x2000e00
@@ -106,7 +106,7 @@
 				cache[node]=True
 				self.pre_recurse(node)
 				try:
-					function_code=node.read('rU',encoding)
+					function_code=node.read('r',encoding)
 					exec(compile(function_code,node.abspath(),'exec'),self.exec_dict)
 				finally:
 					self.post_recurse(node)
@@ -344,9 +344,9 @@
 		return cache_modules[path]
 	except KeyError:
 		pass
-	module=imp.new_module(WSCRIPT_FILE)
+	module=types.ModuleType(WSCRIPT_FILE)
 	try:
-		code=Utils.readf(path,m='rU',encoding=encoding)
+		code=Utils.readf(path,m='r',encoding=encoding)
 	except EnvironmentError:
 		raise Errors.WafError('Could not read the file %r'%path)
 	module_dir=os.path.dirname(path)
--- aubio-0.4.9/waflib/ConfigSet.py.vanilla	2024-05-10 23:32:33.237462724 +0200
+++ aubio-0.4.9/waflib/ConfigSet.py	2024-05-10 23:32:49.004461032 +0200
@@ -146,7 +146,7 @@
 		Utils.writef(filename,''.join(buf))
 	def load(self,filename):
 		tbl=self.table
-		code=Utils.readf(filename,m='rU')
+		code=Utils.readf(filename,m='r')
 		for m in re_imp.finditer(code):
 			g=m.group
 			tbl[g(2)]=eval(g(3))
